packages/genesys-spark-components/src/components/beta/gux-side-panel/components/gux-side-panel-heading/gux-side-panel-heading.tsx

```diff
@@ -0,0 +1,47 @@
+import { Component, Element, h, Prop } from '@stencil/core';
+import { trackComponent } from '@utils/tracking/usage';
+import {
+  GuxSidePanelHeadingLevel,
+  GuxSidePanelHeadingTag
+} from '../../gux-side-panel.types';
+
+@Component({
+  tag: 'gux-side-panel-heading',
+  styleUrl: 'gux-side-panel-heading.scss',
+  shadow: true
+})
+export class GuxSidePanelHeading {
+  @Element()
+  private root: HTMLElement;
+  /**
+   * Heading level, 1-6.
+   */
+  @Prop()
+  level: GuxSidePanelHeadingLevel = 1;
+
+  @Prop()
+  text: string;
+
+  @Prop()
+  icon: string;
+
+  private HeadingTag: GuxSidePanelHeadingTag;
+
+  componentWillLoad(): void {
+    trackComponent(this.root);
+    this.HeadingTag = `h${this.level}` as GuxSidePanelHeadingTag;
+  }
+
+  render(): JSX.Element {
+    return (
+      <this.HeadingTag>
+        {this.icon && (
+          <gux-icon decorative size="medium" icon-name={this.icon} />
+        )}
+        <gux-truncate>
+          <slot />
+        </gux-truncate>
```

Comment:
![Screenshot 2024-11-05 at 10 29 29](https://github.com/user-attachments/assets/093e5c4c-440a-4c98-ac35-96ef87987f36)
The title text is showing the truncation tooltip when it isn't truncated. It looks like the line-height of 1 is a bit too tight for it. I checked the typography page and it recommends 27px for 18px font-size. 

I can't see a clear pattern in the ratio between font-size and line-height so you could hedge it 1.5 maybe.